README.md

```diff
@@ -29,6 +29,19 @@ make run
 
 It will read the contents of `config.yml` and begin its loop. You could also run it directly and provide a new config file via the `--config` argument.
 
+
+#### Docker
+
+You can pull this repository directly from Docker:
+
+`docker pull kevineaton/minitor`
+
+The Docker image uses a default `config.yml` that simply pings a server every 30 seconds with exponential backup. This is likely not what you want, so when you run the Docker image, you should supply your own `config.yml` file:
+
+`docker run -v $PWD/config.yml:/app/config.yml kevineaton/minitor`
+
+Note that the image uses the version of minitor in `pip` and does not build from source.
```

Comment:
It might be worth discussing if this should be building from source or from pip.

Reply:
If its installing from pip its required to build the Docker image AFTER the push to pip is completed. If you build it from source you can just build the Docker Image or build the sources directly and run them.

Reply:
I would be concerned that the version in PIP doesn't equal the version in source, which is why I chose to build from PIP rather than from source. If the docker image is running a different version than the PIP version, documentation, but tracking, etc can become more challenging. If there is a CI tool that essentially builds the docker image AND pushed to pip at the same time, that would alleviate that concern.

Reply:
Docker Hub does allow building of tags as versions. That could be a way of making sure that v0.1.0 == v0.1.0 across pip and docker. Using minitor:latest though would give the current master. Might be nice for someone who wants bleeding edge since I'm not going to push those to pip.

I can also add a build on my Drone.io server to push pip automatically when tagged as well. I can see about setting that up.

For now we'll leave it installing from pip and revisit when some more automation is in place.

Reply:
Yeah - i think its the way literally everone works with docker image or? But i think latest should equal to the latest stable version and develop/bleeding edge could something be like :unstable ... :)

Reply:
I now have automated building for pypi that I'm testing on my Drone server. I also set up a Travis build to just run tests and validate PRs from Github.

Once I'm set up, any tags will auto build and push to pypi. Then we can actually have the Dockerfile install minitor using `pip install -e .` and it can build `master` as `latest` and then build and tag all git tags.

Reply:
I still think that this is not the way to go with Docker. Instead you should just build the docker image on each push to master and on each tag. For example, this is the way im doing it on my docker system images:

```
build-image:
  stage: docker
  image: docker:latest
  except:
  - master
  services:
  - docker:dind
  script:
  - "docker pull java:8-jre-alpine"
  - "docker build -t $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:$CI_COMMIT_REF_NAME -t $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:latest -t $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:stable ."


publish-docker-latest:
  stage: docker-push
  image: docker:latest
  allow_failure: true
  except:
  - master
  services:
  - docker:dind
  script:
  - "docker login -u=$DOCKER_HUB_USER -p=$DOCKER_HUB_PASSWORD"
  - "docker push $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:latest"

publish-docker-develop:
  stage: docker-push
  image: docker:latest
  allow_failure: true
  only:
  - develop
  services:
  - docker:dind
  script:
  - "docker login -u=$DOCKER_HUB_USER -p=$DOCKER_HUB_PASSWORD"
  - "docker push $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:develop"

publish-docker-tag:
  stage: docker-push
  image: docker:latest
  allow_failure: true
  services:
  - docker:dind
  only:
  - tags
  script:
  - "docker login -u=$DOCKER_HUB_USER -p=$DOCKER_HUB_PASSWORD"
  - "docker push $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:$CI_COMMIT_REF_NAME"
  - "docker push $DOCKER_NAMESPACE/$DOCKER_IMAGE_NAME:stable"


```

*Be aware, this example is based of my gitlab-ci file*


In this configuration i can push to develop to update the develop tag on docker-hub.
Pushing to a tag leads to a push to docker hub with the corresponding tag (eg.: 1.11 => 1.11 in docker hub).
Pushing to any (except master) branch leads to a push to latest. (like bleeding edge)

In every push/tag the image is rebuild in the *build-image* step.

Reply:
I believe that the end result is the same as I'm describing.

Every push to master would build the `latest` image and tags would build tagged images for each version.

Reply:
Yeah, but it could get tricky if you create a tag on an older commit, the image will be built with `pip install minitor` and maybe get newer sources/binaries.. I still think its better to just run
```python setup.py build
python setup.py install
```
instead of
```
pip install minitor
```

and continue with the ENTRYPOINT (using the same command as it would if it was installed by pip) just to make sure to get the actual commit sources..

Regards
SeaLife

Reply:
Oh, I see what you mean. `pip install -e .` will install whatever is in the local directory, not pull from pypi. So no risk on getting the wrong version.

It's basically the same as doing a build and install except that it runs using the local copy of code rather than bundling it and installing it elsewhere on the system, leaving multiple copies.